### PR TITLE
Update precommit hooks to work in GitHub Desktop (Port)

### PR DESCRIPTION
**Pull request by Bibby0110**
## Intent of your Pull Request
Ports https://github.com/tgstation/tgstation/pull/51404

Turns out GitHub Desktop actually does bundle a sh.exe, just not a bash.

Also tries to look for the py launcher to help people who didn't check the "Add Python to PATH" option.

### CODE REVIEW

**Comment by JamieD1** [comment]
REEEEEEE I was going to port this

**Comment by alexkar598** [comment]
do we really need to enable github desktop users :(

**Comment by Bibby0110** [comment]
Yes ;)

File: tools/hooks/python.sh [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
+elif command -v py >/dev/null 2>&1; then
+	PY=py
+else
+	echo "Please install Python 3.6 or later."
 fi
 PATHSEP=$($PY - <<'EOF'
 import sys, os
 if sys.version_info.major != 3 or sys.version_info.minor < 6:
-	sys.stderr.write("Python 3.6+ is required: " + sys.version + "\n")
+	sys.stderr.write("Python 3.6 or later is required, but you have:\n" + sys.version + "\n")
```
Comment (ThatLing) [inline tools/hooks/python.sh]:
```suggestion
	sys.stderr.write("Python 3.6 or later is required, but you have: " + sys.version + "\n")
```

**Review by alexkar598 — APPROVED** [review]

**Pull request merged**